File: cluster/sharding/javadsl/src/main/java/play/javadsl/cluster/sharding/typed/ClusterShardingComponents.java
/*
 * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package play.javadsl.cluster.sharding.typed;

import akka.cluster.sharding.typed.javadsl.ClusterSharding;
import play.components.*;
import play.internal.javadsl.cluster.sharding.typed.ClusterShardingProvider;
import akka.annotation.ApiMayChange;

@ApiMayChange
public interface ClusterShardingComponents extends AkkaComponents {

  default ClusterSharding clusterSharding() {
    return new ClusterShardingProvider(actorSystem()).get();
  }
}

File: cluster/sharding/scaladsl/src/main/scala/play/scaladsl/cluster/sharding/typed/ClusterShardingComponents.scala
/*
 * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package play.scaladsl.cluster.sharding.typed

import akka.actor.ActorSystem
import akka.cluster.sharding.typed.scaladsl.ClusterSharding
import akka.actor.typed.scaladsl.adapter._
import akka.annotation.ApiMayChange

@ApiMayChange
trait ClusterShardingComponents {
  def actorSystem: ActorSystem
  lazy val clusterSharding: ClusterSharding = new ClusterShardingProvider(actorSystem).get
}
